### PR TITLE
Remove no-op assert statement in ClientQuerySegmentWalker

### PR DESCRIPTION
**Pull request by suneet-s**
The assert statement in ClientQuerySegmentWalker will always be true because
of the preceding while loop which has the same condition.
This change removes the check to fix the error reported by LGTM

### CODE REVIEW

**Comment by gianm** [comment]
This is kind of a weird flag. The whole point of asserts is that they are supposed to always be true. They are like comments which automatically enforce themselves. I don't see why we should remove one just because LGTM can prove that it's always true.

I'd vote to either suppress the check for this particular instance, or at least replace the assert with a comment like:

```
// At this point we know "current" is not a QueryDataSource.
```

**Comment by suneet-s** [comment]
> This is kind of a weird flag. The whole point of asserts is that they are supposed to always be true. They are like comments which automatically enforce themselves. I don't see why we should remove one just because LGTM can prove that it's always true.
> 
> I'd vote to either suppress the check for this particular instance, or at least replace the assert with a comment like:
> 
> ```
> // At this point we know "current" is not a QueryDataSource.
> ```

I misread the warning - I thought lgtm was complaining that the assert expression was always true, but after reading your comment I paid closer attention to the error. It's actually warning us that `current instanceof QueryDataSource` is always false when what it should be checking is whether `current not instanceof QueryDataSrource` is always false.

I filed a bug for it on lgtm https://github.com/Semmle/ql/issues/3249

Going to hide this warning.

File: server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java [hunk 1]
```diff
@@ -274,8 +274,6 @@ private DataSource inlineIfNecessary(
           current = Iterables.getOnlyElement(current.getChildren());
         }
 
-        assert !(current instanceof QueryDataSource);
-
         current = inlineIfNecessary(current, null, subqueryRowLimitAccumulator, maxSubqueryRows, dryRun);
```
Comment (suneet-s) [inline server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java:277]:
```suggestion
        assert !(current instanceof QueryDataSource); // lgtm [java/contradictory-type-checks]
        
        current = inlineIfNecessary(current, null, subqueryRowLimitAccumulator, maxSubqueryRows, dryRun);
```

File: server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java [hunk 2]
```diff
@@ -274,8 +274,8 @@ private DataSource inlineIfNecessary(
           current = Iterables.getOnlyElement(current.getChildren());
         }
 
-        assert !(current instanceof QueryDataSource);
-
+        assert !(current instanceof QueryDataSource); // lgtm [java/contradictory-type-checks]
+
```
Comment (suneet-s) [inline server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java:278]:
```suggestion

```

**Pull request merged**